### ISSUE TITLE
Strange input problems resulting in chaotic behavior

### ISSUE THREAD

**Issue by Maoman1** [Raicuparta/nomai-vr#307]
* OS: Windows 7 x64
* Game Version: 1.0.7.481
* OWML Version: 0.6.1
* NomaiVR Version: 0.48
* VR hardware: Original Vive headset and lighthouses + Index controllers (also tried original Vive wands, same behavior)
* Log file: [OWML.Log.27-07-2020-22.02.14.txt](https://github.com/Raicuparta/nomai-vr/files/4985561/OWML.Log.27-07-2020-22.02.14.txt)

**What I've tried:**
* Validating SteamVR files
* Enabling "Force OpenVR mode" 
* Enabling "Mod button combinations block game input"
* Disabling "Use Desktop Game Theatre while SteamVR is active"
* Disabling Steam Overlay
* Unplugging all gamepad-like devices and quitting their respective software (PS3 controller, Razer Orbweaver, XP-Pen G640 tablet)
* Creating an entirely new controller binding in SteamVR, deleting every input and replacing them exactly as before
* Possibly other things I'm forgetting, I was starting to get pretty frustrated and had to just put it all down for a couple hours before I made this post.

**Description:**

**In the main menu:**
It's as if the game is reading my inputs as bursts of random button mashing. 

Pulling the trigger or moving the analog stick in any direction does a lot of random stuff all at once. I *believe* what it's doing is attempting to change menu tabs (multiple times in rapid succession judging from the SFX) while simultaneously attempting to select both the highlighted menu option and the menu option being pointed at by the controller laser, with only one of these actions actually succeeding. In other words, pulling the trigger will randomly either: jump several menu tabs over, select the highlighted item, or select what I'm pointing at... but occasionally seems to also jump several menu tabs and THEN select something on that new tab. It's very random and chaotic.

Pressing B *usually* selects the menu item I'm pointing at, but sometimes (maybe one out of four times) selects something unexpected and I can't make sense of the pattern.

If my hand is touching either controller's grip at all, the touchpad does nothing, but if I carefully keep my fingers off the touch-sensitive area of the grip, the touchpad will select "back" as intended. 

**In game:**
Despite the binding saying left stick is move and right stick is look, both analog sticks do the same thing. It's a little weird to explain but I can only move along a clockwise circle, as if the only possible inputs are "walk forward + turn right" or "walk backward + turn left." 

The axis for this movement is aligned diagonally on the stick; i.e. pointing the stick up-left or down-right results in no movement at all, pointing straight up or right results in slow forward + right turn movement, pointing up-right results in fast forward + right turn movement. This movement is always relative to my head even when I change the setting so movement should be relative to my left hand: if I move the analog stick up-right and then physically spin my body counter-clockwise it is possible to walk in a more-or-less straight line, albeit with great difficulty.

However, I would also occasionally get stuck walking forward+right-turn for several seconds despite not touching either analog stick at all. I could not make any sense of what caused this to happen or what made it stop, it was seemingly entirely random.

Oddly enough, all other controls seemed to work perfectly fine. I could interact with the lift and the cockpit seat, I could jump, I could grab and use both the translator and the signalscope and release them, and I could open and close the menu as expected (but the in-game menu had the same buggy behavior as the main menu). I didn't try talking to any NPC.

By doing the whole "turn my body counter-clockwise" thing, I was able to drunkenly stumble my way up the lift, into the ship, and managed to get into the pilot seat, but no controls activated any thrust or the liftoff sequence. Not all controls were ignored though, as I *was* able to toggle the landing camera and exit the pilot seat again.

...

I really hope I'm just missing some simple setting or something and this is a relatively simple fix, because this game is just magnificent and I was very excited when I learned I could play it in VR. Even simply looking around the world while trying to diagnose this buggy behavior was impressive enough.

**Comment by Raicuparta** [comment]
Not sure if the problem is related to the errors showing up in that log file. Seems to happen only in version 1.0.7.481 (on Steam), but not on 1.0.7.463 (Epic). Will have to investigate.

**Comment by Raicuparta** [comment]
@Maoman1 Try this:
Make sure no overlays are open in SteamVR when you first start up the game. Probably also best to make sure you have your headset on your head (so it doesn't go to sleep). Seems like there's some bug where if the game isn't focused while starting up, it throws some errors that I see in your logs file. Those errors might not be what's causing your problem, though.

**Comment by Maoman1** [comment]
Nope, didn't seem to make any difference. Here's some new logs if you wanna check through them 
[OWML.Log.28-07-2020-00.40.19.txt](https://github.com/Raicuparta/nomai-vr/files/4986033/OWML.Log.28-07-2020-00.40.19.txt)

**Comment by Raicuparta** [comment]
> Nope, didn't seem to make any difference. Here's some new logs if you wanna check through them
> [OWML.Log.28-07-2020-00.40.19.txt](https://github.com/Raicuparta/nomai-vr/files/4986033/OWML.Log.28-07-2020-00.40.19.txt)

Looks like the errors are gone. The remaining errors are stuff that happens always. So I'm not sure what else it could possibly be 🤷‍♀️ Other people who use Valve Index controllers seem to have no problem, but I'll confirm to see if they have issues with the latest versions.

**Comment by Maoman1** [comment]
I assume you're talking about the GetOriginTrackedDeviceInfo error. It's weird that didn't show up in that log, because it *does* show up in my next log that's more recent: 
[OWML.Log.28-07-2020-03.37.38.txt](https://github.com/Raicuparta/nomai-vr/files/4986753/OWML.Log.28-07-2020-03.37.38.txt)
 The only thing that changed between the two is I disabled steam overlay, but it's still doing the same behavior.

**Comment by Raicuparta** [comment]
> I assume you're talking about the GetOriginTrackedDeviceInfo error. It's weird that didn't show up in that log, because it _does_ show up in my next log that's more recent:
> [OWML.Log.28-07-2020-03.37.38.txt](https://github.com/Raicuparta/nomai-vr/files/4986753/OWML.Log.28-07-2020-03.37.38.txt)
> The only thing that changed between the two is I disabled steam overlay, but it's still doing the same behavior.

Yes, it's the steam overlay that causes those errors. But they aren't related to your issue, that's something else.

There's really not much more help I can offer. The mod code seems to be doing what I expect it to. If it's some device-specific thing (which we don't even know if it is), then it's hard for me to find the source of the problem, because I can't reproduce it without the device.

For now, all I can hope for is that you accidentally stumble on a fix yourself, and in that case please let me know.

**Comment by Maoman1** [comment]
>Yes, it's the steam overlay that causes those errors

But disabling steam overlay was the last thing I tried, I disabled it and *then* got that latest log I just sent you. You can even see in the file name the timestamp is 3 hours after the previous one that didn't have the errors and before I disabled steam overlay. I understand it's unrelated to my issue, it's just strange that they'd show up after disabling it like that.

>For now, all I can hope for is that you accidentally stumble on a fix yourself, and in that case please let me know.

Wait, really? You don't even have any ideas for other things I could try? I've been trying stuff on and off all day and am entirely out of ideas... stumbling blindly around, trying completely random things without a clue what I'm doing and just hoping for a solution is really quite frustrating.

**Comment by Raicuparta** [comment]
> stumbling blindly around, trying completely random things without a clue what I'm doing and just hoping for a solution is really quite frustrating.

This nicely sums up my experience developing this mod 😅 But yeah, if I can't reproduce problems, I don't have much of a chance at fixing them, that's just how software development generally works. Just noticed now that you tried with two different controllers though, so those don't seem to be problem.

If you're reeeeally dedicated to finding a solution to this, one thing I can suggest is trying the Epic version. Epic version doesn't have Steam DRM bs and has been much easier to work with in general. This is also usually the version you'll find if you sail the seven seas, since it has no DRM. You can even just buy it on Epic, copy the files somewhere else, and the refund it, since there's no DRM the game works. You bought the game already so who cares 🤷🏻‍♂️ Just make sure you have version 1.0.7 or it won't work.

**Comment by Maoman1** [comment]
Lol alright then... inb4 it still happens with the ~~seven seas edition~~ epic version.

**Comment by Maoman1** [comment]
Hey guess what!

...Same problem. 
[OWML.Log.28-07-2020-05.58.50.txt](https://github.com/Raicuparta/nomai-vr/files/4987475/OWML.Log.28-07-2020-05.58.50.txt)

**Comment by Maoman1** [comment]
Hey guess what!

...Same problem.
[OWML.Log.28-07-2020-06.08.19.txt](https://github.com/Raicuparta/nomai-vr/files/4987537/OWML.Log.28-07-2020-06.08.19.txt)

**Comment by Tiergott** [comment]
I have exactly the same problems on my system - tried everything, that was mentioned here and no luck - all issues are reproducing.
[Logs.zip](https://github.com/Raicuparta/nomai-vr/files/5006195/Logs.zip)

By the way - I have Samsung Odyssey+, win10 pro version 2004, 1080 Ti, I7 7700k, steamvr beta v.1.14.3 (1596060603)

**Comment by Tiergott** [comment]
Seems I manage to fix issues by:

1. Create a new profile in game
2. Switch to it
3. Restart the game and start from begininng

Yes - I lost previous nonvr saves, but I played only for half an hour - so - no big deal.

This game is awesome in vr. If only optimisation was better.

**Comment by Raicuparta** [comment]
@Tiergott Can you confirm that if you go to the previous profile, the problem comes back? I'm wondering why profile would affect it. Perhaps because of different controller bindings in your other profile.

**Comment by Tiergott** [comment]
I confirm, that if I switched to previous profile - the problem comes back. And if I came back to new one - all is fine.
Also - I did not change any controller mappings on previous profile

**Comment by Raicuparta** [comment]
> I confirm, that if I switch to previous profile - the problem comes back. And if I came back to new one - all is fine.
> Also - I did not change any controller mappings on previous profile

Awesome, that makes me happy. Reproducible errors are the best kind. Any chance you played with a controller before? Maybe a playstation controller or something else that's not an Xbox controller?

**Comment by Tiergott** [comment]
I strongly agree with you regarding reproducible errors.
I have both ps4 and xbox360 controllers, but I did not use them in game, if I remember correctly.
Maybe this issue appears because of some progress in game itself?...

**Comment by Raicuparta** [comment]
@Tiergott can you do me a favor? go to the Outer Wilds save directory (something like  `C:\Users\Tiergott\AppData\LocalLow\Mobius Digital\Outer Wilds`). Make a zip of the whole thing and send it to me somehow (ex. upload to Google Drive and share link, or whatever other way you prefer). This way I can check exactly what's causing the problem.

**Comment by Tiergott** [comment]
[Outer Wilds.zip](https://github.com/Raicuparta/nomai-vr/files/5010046/Outer.Wilds.zip)

No problem - here is an archive

**Comment by Raicuparta** [comment]
Thanks for the help @Tiergott , I think I was able to fix this in a way that does't happen to other people. So you can rest assured that future players won't have to suffer the way you have.

**Comment by Tiergott** [comment]
Great! I am glad to help:)

**Comment by Raicuparta** [comment]
Reopening this since it suddenly happened to me. Seems like my fix didn't actually fix all scenarios.

**Comment by Maoman1** [comment]
I saw all the updates here and tried again with a completely fresh install, and I have good news and bad news:

Creating a new profile in game and starting over DID fix the issue (omg thank you Tiergott) and I am finally able to experience this game in VR. 

However it did still happen _until_ I made a new profile, so it doesn't seem that this is fixed so it "doesn't happen to other people."

**Comment by Raicuparta** [comment]
I'll reopen since it seems like it wasn't fixed for all cases (although I can confirm that it is fixed in some specific scenarios). Unlikely that I will ever try to fix this again, though, seems like there are too many scenarios where it can be triggered.
